From c1dd524816be4cc2072be38cd72715a9684ecf56 Mon Sep 17 00:00:00 2001
From: Emil Jahshan <emil.jahshan@intel.com>
Date: Thu, 7 Oct 2021 11:22:12 +0300
Subject: [PATCH] d4xx: override depth DT in MIPI header

 - Custom DT 0x31 is not supported in Jetson TX2 kernel,
   thus we override the DT in MIPI header to 0x1E

Signed-off-by: Emil Jahshan <emil.jahshan@intel.com>
---
 drivers/media/i2c/d4xx.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 836c453bd..76c5a6efd 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -879,6 +879,14 @@ static int ds5_configure(struct ds5 *state)
 		return ret;
 
 	if (depth->streaming) {
+		// TODO: remove if 0x31 DT is supported
+		//       this will configure FW to override DT in mipi header to
+		//       0x1E for depth, since DT 0x31 is not supported
+		//       in Jetson
+		ret = ds5_write(state, 0x401C, 0x1E);
+		if (ret < 0)
+			return ret;
+
 		ret = ds5_write(state, DS5_DEPTH_Y_FPS, depth->config.framerate);
 		if (ret < 0)
 			return ret;
-- 
2.17.1

